pkg/apis/pipeline/v1alpha1/pipelinerun_validation.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	"k8s.io/apimachinery/pkg/api/equality"
+)
+
+// Validate pipelinerun
```

Comment:
just as a general comment, I've had mixed feelings about the linter's warnings about these functions not having comments - comments which restate what the function does dont seem super valuable

i wonder if we could add more value by including some more information - e.g. what it means for a pipelinerun to be valid, and/or when we'd expect this to be called 

or could we make some of these functions private?